velox/exec/fuzzer/DuckQueryRunner.cpp

```diff
@@ -341,38 +373,53 @@ std::optional<std::string> DuckQueryRunner::toSql(
   return sql.str();
 }
 
-std::optional<std::string> DuckQueryRunner::toSql(
-    const std::shared_ptr<const core::HashJoinNode>& joinNode) {
-  const auto& joinKeysToSql = [](auto keys) {
-    std::stringstream out;
-    for (auto i = 0; i < keys.size(); ++i) {
-      if (i > 0) {
-        out << ", ";
-      }
-      out << keys[i]->name();
+static const std::string joinKeysToSql(
+    const std::vector<core::FieldAccessTypedExprPtr>& keys) {
+  std::stringstream out;
+  for (auto i = 0; i < keys.size(); ++i) {
+    if (i > 0) {
+      out << ", ";
     }
-    return out.str();
-  };
+    out << keys[i]->name();
+  }
+  return out.str();
+}
 
-  const auto filterToSql = [](core::TypedExprPtr filter) {
-    auto call = std::dynamic_pointer_cast<const core::CallTypedExpr>(filter);
-    return toCallSql(call);
-  };
+static std::string filterToSql(const core::TypedExprPtr& filter) {
+  auto call = std::dynamic_pointer_cast<const core::CallTypedExpr>(filter);
+  return toCallSql(call);
+}
 
-  const auto& joinConditionAsSql = [&](auto joinNode) {
-    std::stringstream out;
-    for (auto i = 0; i < joinNode->leftKeys().size(); ++i) {
-      if (i > 0) {
-        out << " AND ";
-      }
-      out << joinNode->leftKeys()[i]->name() << " = "
-          << joinNode->rightKeys()[i]->name();
+static std::string joinConditionAsSql(const core::AbstractJoinNode& joinNode) {
+  std::stringstream out;
+  for (auto i = 0; i < joinNode.leftKeys().size(); ++i) {
+    if (i > 0) {
+      out << " AND ";
     }
-    if (joinNode->filter()) {
-      out << " AND " << filterToSql(joinNode->filter());
+    out << joinNode.leftKeys()[i]->name() << " = "
+        << joinNode.rightKeys()[i]->name();
+  }
+  if (joinNode.filter()) {
+    if (!joinNode.leftKeys().empty()) {
+      out << " AND ";
     }
-    return out.str();
-  };
+    out << filterToSql(joinNode.filter());
+  }
+  return out.str();
+};
+
+std::optional<std::string> DuckQueryRunner::toSql(
+    const std::shared_ptr<const core::HashJoinNode>& joinNode) {
+  std::string probeTableName;
+  std::string buildTableName;
+  if (const auto probeSubQuery = toSql(joinNode->sources()[0]),
```

Comment:
nit: maybe pull the initializations out of the if and leave the condition as `if (probeSubQuery && buildSubQuery)` for readability?

Reply:
Done.